lib/storage/index_db.go

```diff
@@ -1046,10 +1046,10 @@ func (is *indexSearch) searchLabelValuesWithFiltersOnDate(qt *querytracer.Tracer
 	if err != nil {
 		return err
 	}
-	if filter != nil && filter.Len() < 100e3 {
+	if filter != nil && filter.Len() < 100e3 && labelName != "" {
```

Comment:
This fix results in a slow path when searching for metric names, e.g. when `labelName=""`. The correct fix is to use `__name__` instead of empty label name inside the `getLabelValuesForMetricIDs()`

Reply:
Tx! Updated